### DISCUSSION TITLE
Source brand-data from WikiData

### DISCUSSION THREAD

**Discussion by Unkn0wnCat** [organicmaps/organicmaps#1052] [Ideas]
***Update: 2022-01-18***

I've created a meta-issue to discuss implementation of this feature as well es some other similar issues. Please participate in #1892 to discuss implementation or work on this!

Have a nice day,
Kevin

---

Hi all,

OpenStreetMap has an ongoing colaboration with the OpenDatabase-project WikiData, for sourcing data like brand information.

I'd love to see Organic Maps get a brand search with Logos pulled from WikiData, as described in the [OSM-wiki page about the brand:wikidata-tag](https://wiki.openstreetmap.org/wiki/Key:brand:wikidata). This could also be used to enable search by [operator, public transport network, memorial subject, or even architect](https://wiki.openstreetmap.org/wiki/Key:wikidata#Secondary_Wikidata_links).
I really think using data and logos from WikiData could make Organic Maps even more lively.

I'm looking forward to hearing your thoughts about this.

Have a nice day,
Kevin

**Comment by lucumon** [comment]
I also asked for this for OsmAnd and they rejected It because being an offline-maps app, this would increase the size of the maps a lot. Maybe here we are at the same situation.

**Reply by Outlet2048** [reply]
I really like that idea since it improves the ease of use and the feature is on par with what google maps has to offer.
Personally, I definitely wouldnt mind the increased storage usage.

The icons wouldnt even need to be that large so I'm not sure how much more storage would be required

**Reply by Unkn0wnCat** [reply]
Especially with SVG files the size difference would not be that big I think. In many ways it's the same as just adding more Metadata to the OSM elements themselves.

But I agree that file size is a thing which would need to be watched, especially in big cities.

One solution may be to allow for "Additional Data Packs" to be downloaded to supplement existing maps, so users can opt-in or opt-out of Wikidata Infos. What do you think about that?

**Reply by Outlet2048** [reply]
I like the idea of additional data packs but it would have to be clear to the end user what exactly is being downloaded.
Considering that it would be optional, why not add a short excerpt of wikipedia in the users language as well? (Applies to POIs with a linked Wikipedia article)

**Reply by Martinligabue** [reply]
Brand data shouldn't be too much of a problem as there are a limited number of brands and the information is the same for each, so it would be a separate file dedicated to that information only, maybe even optionally downloadable.

**Comment by Raphael0** [comment]
Hi, though I think this is a nice idea, I am personally strongly against it as it would automatically create a distinction between larger brands and more traditional stores, making brands stand out more. I very much like the current implementation treating them equally.

Have a nice day too

**Reply by Unkn0wnCat** [reply]
I see your point, but I'd separate this into two parts.

Part 1 would be enabling "Search by Brand" where if you'd type "Burger King" into the search it would directly filter by the WikiData-ID in the brand field. I think this would not be hurtful to smaller stores, as it would only improve sorting and quality of the search results.

Part 2 would be bringing brand identity onto the map. This is where I see the issue you mentioned. I agree that showing colorful logos on the map would draw the user to those. However I think this might be acceptable in certain categories, such as gas stations, where many people have cards for a specific gas station chain, and might want to quickly recognize which gas station they can stop at. However for stuff like restaurants I can see this being a problem.

So my solution would be to only whitelist certain categories for on-map-logos (for example gas stations), whilst keeping restaurants and others as the default icon.

I'm not sure about how hurtful showing logos in search would be in contrast to the improvement in UX. What do you think?

**Reply by Outlet2048** [reply]
The above mentioned solution would fix the issue because looking up Burger King means that I'm not looking for traditional stores. Therefore, this implementation wouldn't hurt their business 

Personally, I dont even mind brands being displayed on the map because it gives me a better overview of where I can find what. Seeing more detail at a glance.

And lastly it helps you find traditional stores when you can just ignore all the POIs with the Burger King logo and vice versa

**Reply by Raphael0** [reply]
You are right, these are two different things.

Adding the brand field shouldn't be problematic (though isn't usually the Brand ID also the POI name? When searching for Burger King, Burger Kings appear in the results)

For the second part, I think this one really comes down to personal preferences. It certainly would be a slight UX improvement and might make recognising some stores etc easier, though it opens the door to some uneasy decisions about which brands are deemed most known/popular/important with a bias for western/international brands.
Also, and this is of course my personal opinion, I see brand logos as a form of advertising that I would like to keep away from my map.

**Reply by Unkn0wnCat** [reply]
The Brand ID and POI name sometimes differ, such as for a bakery chain around my hometown. Also sometimes the search goes a bit wonky trying to figure out the relevance of the same brand store over others - especially if the brand name is not typed in 100% correctly ("McDonalds" vs. "McDonald's"), so recognizing it is a brand being searched and just sorting the results by distance would definitely help with this.

As for the logos on the map, you're right this absolutely is a personal preference, and as such, if added, should be customizable by the user. I didn't think of that. Maybe some kind of toggle to toggle between "reasonable defaults" and off, with an additional menu for users to check or uncheck categories for which logos will be shown? Would that be too much overhead for a feature like this?

**Comment by ghost** [comment]
This feature would be very useful for Organic Maps. I understand the reasoning that you don't want to advertise big brands, giving them unfair advantages over smaller, traditional shops. However, there are some shop categories that this concern is not applicable and would certainly be useful to see their logo on the map or make searching easier:
- banks
- gas stations
- mobile_phone shops (for example Vodafone, Deutsche Telekom, AT&T, etc.)
- energy_supplier shops
- post_office or courier shops

One thing that I'd like to point out is that you'll have to take into account popular brands by country. Showing only world-wide or US brands only would be pointless since not all multinational companies are active in every country. I don't know how you're going to be implementing this but certainly you could simply take every brand that ID editor has in its database and include it in your app.

**Reply by biodranik** [reply]
Are there any other maps apps that already implemented brands icons? It would be great to see some screenshots and estimate the usability.

**Reply by Outlet2048** [reply]
![Screenshot_20210830-134929.jpg](https://user-images.githubusercontent.com/72819077/131334768-fb214874-8897-47e5-9c87-be3068a3a7ac.jpg)

Magic Earth shows brand icons when you specifically search for a brand.
The same applies to browsing the map:
![Screenshot_20210830-135126.jpg](https://user-images.githubusercontent.com/72819077/131334927-dd3e77f4-8a92-409e-9236-37d45d9b5d75.jpg)

Google maps shows the icons when searching for a brand
![Screenshot_20210830-135020.jpg](https://user-images.githubusercontent.com/72819077/131334963-f799e456-41d9-40b4-90ec-a524d0c51710.jpg)

Personally, I prefer the way magic earth does it.

**Reply by Outlet2048** [reply]
![Screenshot_20210830-170800.jpg](https://user-images.githubusercontent.com/72819077/131367074-1d753948-44b9-4b3e-8e63-213da8157ac6.jpg)

Here's another screenshot that I took a few minutes ago
That's what brand icons look like while driving / in routing mode

**Comment by dbf256** [comment]
These logos are all in different styles and are too colorful. Map full of them looks like http://www.milliondollarhomepage.com/ The examples that are displayed above are awful comparing to the current map style. I hope that it will not be implemented.
Even if I search for some shop/brand name I'm not interested in displaying it's logos in search, name is enough.
We had internet full of banners and billboards full of logos and ads. We should not replicate it in maps application.

**Reply by Outlet2048** [reply]
The organic maps icons are all round and smaller ones when compared to the other apps. So the implementation as shown on the Screenshots wouldn't work as well in organic maps.

Rounded brand icons would work better for organic maps. Sadly I don't have the ability to create a mockup. 
But making branded icons optional shouldn't affect users who don't want to see them

**Comment by vachan-maker** [comment]
The sourcing of brand data is a great idea but I am also interested in Organic Maps displaying images, a short description and other details about a particular POI. 
Currently, Organic Maps mainly displays phone numbers, websites, etc of a POI and only a wikipedia link is show. Pulling more information from wikidata will be helpful to users.

**Reply by Outlet2048** [reply]
Descriptions could be pulled from either wikidata or the description tag from osm data

Either option would be appreciated

**Reply by biodranik** [reply]
Is there already an issue to add OSM description tag?

**Reply by vachan-maker** [reply]
Issue  #1233?

**Reply by Outlet2048** [reply]
I think this only talks about the time when you create a new POI

**Reply by biodranik** [reply]
Then it makes sense to create an issue for Descriptions tag support. After implementing it in generator and apps UI, we can add into the description field any additional info from any sources, including Wikimedia.

**Reply by Outlet2048** [reply]
https://github.com/organicmaps/organicmaps/issues/1888

Here's a generic draft to keep track of it

**Comment by yotamN** [comment]
I think logo is the least interesting data you can extract from Wikidata. Sure, it gives a nice visualization for the store but can look too colorful and give an advantage to big chains as people had already said.

What I think is more interesting is pulling data like social accounts and website from there. There are POIs without this data but the data exist in Wikidata so it would be useful to fetch it.
